GitUI/CommandsDialogs/FormCherryPick.cs

```diff
@@ -45,94 +46,112 @@ private void Form_Closing(object sender, FormClosingEventArgs e)
 
         private void LoadSettings()
         {
-            AutoCommit.Checked = AppSettings.CommitAutomaticallyAfterCherryPick;
-            checkAddReference.Checked = AppSettings.AddCommitReferenceToCherryPick;
+            cbxAutoCommit.Checked = AppSettings.CommitAutomaticallyAfterCherryPick;
+            cbxAddReference.Checked = AppSettings.AddCommitReferenceToCherryPick;
         }
 
         private void SaveSettings()
         {
-            AppSettings.CommitAutomaticallyAfterCherryPick = AutoCommit.Checked;
-            AppSettings.AddCommitReferenceToCherryPick = checkAddReference.Checked;
+            if (DialogResult == DialogResult.OK)
```

Comment:
I would not add this condition. If changed, the user *has* clicked the CheckBoxes.
Sometimes, I need to close a dialog in order to lookup a detail and come back...

Reply:
Closing the dialog is essentially aborting the process. In this case, it doesn't feel right to retain the selection.